core-common/src/main/java/org/mapstruct/Context.java

```diff
@@ -135,7 +136,55 @@
  * }
  * </code>
  * </pre>
+ * <p>
+ * <strong>Example 3:</strong> Using {@code @Context} parameters for creating an entity object by calling an
+ * {@code @}{@link ObjectFactory} methods:
+ *
+ * <pre>
+ * <code>
+ * // type of the context parameter
+ * public class ContextObjectFactory {
+ *     &#64;PersistenceContext(unitName = "my-unit")
+ *     private EntityManager em;
+ *
+ *     &#64;ObjectFactory
+ *     public Valve create( String id ) {
+ *        Query query = em.createNamedQuery("Valve.findById");
+ *        query.setParameter("id", id);
+ *        Valve result = query.getSingleResult();
+ *        if ( result != null ) {
+ *            result = new Valve( id );
+ *        }
+ *        return result;
+ *     }
+ *
+ * }
+ *
+ * &#64;Mapper
+ * public interface ContextWithObjectFactoryMapper {
+ *     Valve map(ValveDto dto, &#64;Context ContextObjectFactory factory);
+ * }
  *
+ *
+ * // generates:
+ *
+ * public class ContextWithObjectFactoryMapperImpl implements ContextWithObjectFactoryMapper {
+ *
+ *   &#64;Override
+ *   public Valve map(ValveDto dto, ContextObjectFactory factory) {
+ *       if ( dto == null ) {
+ *           return null;
+ *       }
+ *
+ *       Valve valve = factory.create();
```

Comment:
In the declaration above, the create method has a String parameter 🙂

Reply:
oops.. will fix.. :smile: